### PR TITLE
Deploy function apps in GitHub deploy actions

### PR DESCRIPTION
**Pull request by nickclyde**
Resolves [this ticket](https://reportstream-service-request.clickup.com/t/36mfw61)

This adds a step to bundle and deploy our main python function app functions in the deploy GitHub Actions. Also fixes a few storage account URLs to use the correct SA name in our Skylight env (piskylightdatasa1)

### CODE REVIEW

**Comment by celestemespinoza** [comment]
Task linked: [CU-36mfw61 Deploy function apps via Terraform](https://app.clickup.com/t/36mfw61)

File: operations/app/terraform/modules/data_factory/linked_services.tf [hunk 1]
```diff
@@ -24,7 +24,7 @@ resource "azurerm_data_factory_linked_service_azure_blob_storage" "pdi_datasa" {
   data_factory_id          = azurerm_data_factory.pdi.id
   integration_runtime_name = azurerm_data_factory_integration_runtime_azure.pdi.name
   use_managed_identity     = true
-  service_endpoint         = "https://${var.resource_prefix}datasa.blob.core.windows.net"
+  service_endpoint         = "https://${var.resource_prefix}datasa${var.environment == "skylight" ? "1" : ""}.blob.core.windows.net"
```
Comment (DanPaseltiner) [inline operations/app/terraform/modules/data_factory/linked_services.tf:27]: This is neat syntax. Was there already a `piskylightdatasa` storage account?

**Review by DanPaseltiner — COMMENTED** [review]
LGTM.

**Review by DanPaseltiner — APPROVED** [review]
LGTM

File: operations/app/terraform/modules/data_factory/linked_services.tf (hunk 1 above)
Comment (nickclyde) [reply operations/app/terraform/modules/data_factory/linked_services.tf:27]: We had one in the other "Skylight PHDI" tenant. I deleted it, but for some reason when I tried to recreate it in the new tenant, it said the name was already taken. 🤷

**Comment by nickclyde** [comment]
Finally got a successful function app deploy!
https://github.com/CDCgov/prime-public-health-data-infrastructure/runs/7366048475?check_suite_focus=true

**Pull request merged**